src/test/java/com/microsoft/sqlserver/jdbc/unit/lobs/lobsStreamingTest.java

```diff
@@ -0,0 +1,249 @@
+package com.microsoft.sqlserver.jdbc.unit.lobs;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.NClob;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.IntStream;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.util.RandomUtil;
+
+
+@RunWith(JUnitPlatform.class)
+public class lobsStreamingTest extends AbstractTest {

```

Comment:
Can you also add a test case that actually surfaces the original NPE problem? I ran this test file against dev branch, and the test doesn't seem to test the NPE case.